### PR TITLE
Exchange base classes updates for FetchCurrencies = emulated & better capability outputs

### PR DESCRIPTION
**Pull request by ttodua**
as title says.

### CODE REVIEW

**Comment by ttodua** [comment]
@kroitor, also, please revise the columns , like `fetchIndexOHLVC` or others. I am not fully clear on this, but maybe some of them could be outdated (of no interest anymore)?

**Comment by ttodua** [comment]
about colors: in Windows `darkred` doesn't work at all and `red.dim` has same color as 'red'. there was `lightRed`  but was nearly undistinguishable from red. In windows (with my tested colors) the only good visual differentiation could be made with `lightMagenta`(as `magenta` doesn't work either) and `red`.

**Comment by kroitor** [comment]
@ttodua sure! Thx! We have actually reviewed them not so long ago, and all those methods should be ok. Methods like `fetchIndexOHLCV`, `fetchMarkOHLCV` are specific to contract markets having "index" and "mark" prices.

File: python/ccxt/async_support/base/exchange.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ async def load_markets_helper(self, reload=False, params={}):
                     return self.set_markets(self.markets)
                 return self.markets
         currencies = None
-        if self.has['fetchCurrencies']:
+        if self.has['fetchCurrencies'] == true:
```
Comment (kroitor) [inline python/ccxt/async_support/base/exchange.py:183]: Check these linting issues: https://app.travis-ci.com/github/ccxt/ccxt/builds/243221395#L1272

It has to be capitalized boolean `True` in Python.

**Comment by kroitor** [comment]
> about colors: in Windows `darkred` doesn't work at all and `red.dim` has same color as 'red'. there was `lightRed` but was nearly undistinguishable from red. In windows (with my tested colors) the only good visual differentiation could be made with `lightMagenta`(as `magenta` doesn't work either) and `red`.

Yes, I think lightMagenta will work.

**Comment by ttodua** [comment]
@kroitor done. all of exchanges amended. 
the only one last remains : okex3. 
if you can check this comment: https://github.com/ttodua/ccxt/blob/exchange-fetchCurrencies/js/okex3.js#L982

and if you see okex3 is heavily used, then maybe contact their api team to make a 'quick-ask' to fix that issue, so 'account/currencies' did not require authorization.

File: js/cdax.js [hunk 2]
```diff
@@ -30,7 +30,6 @@ module.exports = class cdax extends Exchange {
                 'createOrder': true,
                 'fetchBalance': true,
                 'fetchClosedOrders': true,
-                'fetchCurrencies': false,
```
Comment (kroitor) [inline js/cdax.js:33]: This edit breaks CDAX:

- https://app.travis-ci.com/github/ccxt/ccxt/builds/243226878#L2904

**Comment by kroitor** [comment]
We must also fix the tests:

- https://github.com/ccxt/ccxt/blob/master/js/test/Exchange/test.fetchCurrencies.js#L18

File: js/cdax.js (hunk 2 above)
Comment (kroitor) [reply js/cdax.js:33]: We most likely have to remove or rewrite fetchCurrencies in CDAX.

Comment (kroitor) [reply js/cdax.js:33]: The reason for this is that CDAX and Huobi JP were both inherited from Huobi initially, and then split into separate exchanges.

**Comment by ttodua** [comment]
@samgermain  hey.  probably our changes will not conflict right? : https://github.com/ccxt/ccxt/pull/10834

**Comment by ttodua** [comment]
@kroitor whenever it will be feasible for you, might be good this PR was merged, otherwise as I've above commented my and @samgermain's commits might be cross-over with conflicting changes.

**Pull request merged**